### PR TITLE
SIDE-2191 - expose update function/ make onClose optional

### PR DESCRIPTION
**Pull request by scottkipferOlive**
This PR:
-  exposes the `whisper.update` function
- make `onClose` optional when creating a whisper.

### CODE REVIEW

**Review by cshields-oliveai — APPROVED** [review]
Only recommendation I could make here is to add a new test to self-test-loop that includes update.

File: ldk/javascript/src/types/oliveHelps/index.d.ts [hunk 1]
```diff
@@ -290,7 +290,7 @@ declare namespace OliveHelps {
   interface Whisper {
     id: string;
     close: Readable<undefined>;
-    // update(whisper: NewWhisper, cb: (err: string) => void): void // TODO: Implement
+    update(whisper: UpdateWhisper, cb?: (err: Error) => void): void
```
Comment (cshields-oliveai) [inline ldk/javascript/src/types/oliveHelps/index.d.ts:293]: Given the current WIP implementation for "form like behavior" for whispers, the LDK will need a way to intercept components given as `UpdateWhisper` here in order to keep track of their value states.

File: ldk/javascript/.eslintrc.js [hunk 2]
```diff
@@ -20,6 +20,8 @@ module.exports = {
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md
     'no-shadow': 'off',
     '@typescript-eslint/no-shadow': ['error'],
+    "no-use-before-define": "off",
+    "@typescript-eslint/no-use-before-define": ["error"]
```
Comment (richseviora) [inline ldk/javascript/.eslintrc.js:24]: I'd much rather suppress the rule in a given instance unless we have a good reason to suppress it overall.

**Review by richseviora — CHANGES_REQUESTED** [review]
Minor change to rule setting - otherwise looks straightforward!

File: ldk/javascript/.eslintrc.js (hunk 2 above)
Comment (richseviora) [reply ldk/javascript/.eslintrc.js:24]: Nope this was my error b/c I didn't read the change closely!

**Review by richseviora — APPROVED** [review]

**Pull request merged**